plugins/woocommerce-blocks/assets/js/blocks/product-template/utils.tsx

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * It find the parent block "woocommerce/product-collection" and return its attributes.
+ */
+export const useProductCollectionBlockAttributes = ( clientId: string ) => {
+	return useSelect(
+		( select ) => {
+			const { getBlockParents, getBlockName, getBlockAttributes } =
+				select( 'core/block-editor' );
+
+			let currentBlockId = clientId;
+			while ( currentBlockId ) {
+				const parentIds = getBlockParents( currentBlockId, true );
```

Comment:
Alternatively we could use [getBlockParentsByBlockName](https://developer.wordpress.org/block-editor/reference-guides/data/data-core-block-editor/#getblockparentsbyblockname) to simplify this.

Reply:
TIL. Using `getBlockParentsByBlockName` really does make more sense. I'll go ahead and make that change.